gcs-sink-connector/src/main/java/io/aiven/kafka/connect/common/config/validators/ClassValidator.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2024 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.common.config.validators;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+
+public class ClassValidator implements ConfigDef.Validator {
+    private final Class<?> baseClass;
+
+    public ClassValidator(final Class<?> baseClass) {
+        this.baseClass = baseClass;
+    }
+
+    @Override
+    public void ensureValid(final String name, final Object value) {
+        if (value != null && !baseClass.isAssignableFrom((Class<?>) value)) {
+            throw new ConfigException(name, value, "must be a subclass of " + baseClass.getName());
+        }
+    }
+}
```

Comment:
Can we instead bring [this](https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/blame/acccbb0fd02a66af4b20ffe36b836cd42327048f/commons/src/main/java/io/aiven/kafka/tieredstorage/config/validators/Subclass.java) validator implementation here? IMO it makes code a little more readable when you write `Subclass.of(SomeClass.class)` rather then `new ClassValidator(SomeClass.class)`

Reply:
Is there a common library that you use for these. It seems copying and duplication is a bad way forward, particularly for a common concept like validation

Reply:
There is no common library yet and when I created the validator I'm suggesting I did not manage to find a ready to use implementation.
I think for now copying is good enough but in future I think if this pattern will repeat we will consider creating our own commons library to address this.